packages/discovery-react-components/src/components/DocumentPreview/components/SimpleDocument/SimpleDocument.tsx

```diff
@@ -23,20 +23,38 @@ interface Props {
    * Callback which is invoked with whether to enable/disable toolbar controls
    */
   setHideToolbarControls?: (disabled: boolean) => void;
-
+  /**
+   * React component rendered as a fallback when no preview is available.
+   * When specified, the default error component which displays `cannotPreviewMessage`
+   * won't be displayed.
+   */
+  fallbackComponent?: ComponentType<ErrorComponentProps>;
+  /**
+   * Error title displayed when no preview can be displayed by this component.
+   * Unused when `fallbackComponent` is provided
+   */
   cannotPreviewMessage?: string;
+  /**
+   * Error message displayed when no preview can be displayed by this component.
+   * Unused when `fallbackComponent` is provided
+   */
   cannotPreviewMessage2?: string;
   loading: boolean;
   hideToolbarControls: boolean;
 }
 
+interface ErrorComponentProps {
+  document?: QueryResult;
```

Comment:
Since `document` is required for `SimpleDocument`'s props, it should be required here, too.
```suggestion
  document: QueryResult;
```